### DISCUSSION TITLE
Second server used and does it scale?

### DISCUSSION THREAD

**Discussion by Chxsta** [qruto/laravel-wave#34] [Q&A] [answered]
Does this implementation of SSE use a second server for handling the connections (e.g. like with Laravel WebSockets)?
Are there any insights of how many concurrent connections can be reached?

**Comment by slavarazum** [comment, marked as answer]
Wave uses the standard Laravel Redis Broadcasting driver, so its scalability should be in line with this. While I've not yet tested large-scale scenarios, I plan to do so soon to verify these expectations.

**Reply by Chxsta** [reply]
Thank you for the quick response.

I'm interested in the implementation details as I recently implemented SSE in Laravel, and it was very inefficient. To keep the SSE connection open, I used a while(true) loop, which consumed a lot of resources on the PHP server.

So, if I understand correctly, in Wave, during the SSE connection setup, a Redis subscribe takes place, and the events published on a Redis channel are sent to the client via SSE. If that's the case, I'm curious about how this works, as Redis subscribe is a blocking operation, and PHP doesn't support multithreading.

How does Wave ensure that the SSE connection is kept open?

**Reply by slavarazum** [reply]
When a new event is published to the Redis channel and the connection is already terminated, the script initiates a shutdown.
So, while Redis awaits a new event, it remains unaware of whether the connection is still open.
